packages/frontend/src/page/document_breadcrumbs.tsx

```diff
@@ -0,0 +1,77 @@
+import { For, Show, createResource } from "solid-js";
+import type { AnalysisDocument, LiveAnalysisDocument } from "../analysis";
+import { getLiveDoc, useApi } from "../api";
+import type { DiagramDocument, LiveDiagramDocument } from "../diagram";
+import type { LiveModelDocument, ModelDocument } from "../model";
+import { assertExhaustive } from "../util/assert_exhaustive";
+import "./document_breadcrumbs.css";
+
+type AnyDocument = ModelDocument | DiagramDocument | AnalysisDocument;
+type AnyLiveDocument = LiveModelDocument | LiveDiagramDocument | LiveAnalysisDocument;
+type AnyDocumentWithRefId = {
+    document: AnyDocument;
+    refId: string;
+};
+
+export function DocumentBreadcrumbs(props: {
+    document: AnyLiveDocument;
+}) {
+    const [documentChain] = createResource(() => props.document, getDocumentChain);
+
+    return (
+        <div class="breadcrumbs-wrapper">
+            <Show when={documentChain()} fallback={<span>Loading...</span>}>
```

Comment:
As in #463, I want to insist that we don't show the "Loading..." text as a fallback. In this case, I think we can just leave it blank while loading.